tests/test/adremover-loggedin-erc20.test.js

```diff
@@ -0,0 +1,201 @@
+
+describe('The Unlock Ad Remover Paywall (logged in user)', () => {
+  beforeAll(async () => {
+    // We first need to deploy the locks
+    const addresses = []
+    // this cannot be done in parallel, or the same lock creation form gets populated with all of them at once
+    addresses.push(
+      (await dashboard.deployLock(
+        locks[0].name,
+        locks[0].expirationDuration,
+        '1000',
+        locks[0].keyPrice,
+        true /* useERC20 */
+      )).toLowerCase()
+    )
+    addresses.push(
+      (await dashboard.deployLock(
+        locks[1].name,
+        locks[1].expirationDuration,
+        '1000',
+        locks[1].keyPrice,
+        true /* useERC20 */
+      )).toLowerCase()
+    )
+    addresses.push(
+      (await dashboard.deployLock(
+        locks[2].name,
+        locks[2].expirationDuration,
+        '1000',
+        locks[2].keyPrice,
+        true /* useERC20 */
+      )).toLowerCase()
+    )
+
+    // save the lock address to pass it to the ad remover paywall
+    addresses.forEach((address, i) => (locks[i].address = address))
+
+    lockSelectors = addresses.map(lock => path =>
+      `[data-address="${lock}"] ${path}`
+    )
+    //debug.debugPage(page, true)
+    const testUrl = url.paywall(
+      `/static/adremover/integrationtesting-loggedin.html?locks=${encodeURIComponent(
+        JSON.stringify(locks)
+      )}&paywall=${encodeURIComponent(
+        url.paywall('')
+      )}&provider=${encodeURIComponent(url.provider)}&logindelay=0`
+    )
+    await page.goto(testUrl, { waitUntil: 'networkidle2' })
+  })
+
+  it('should load the data iframe and checkout iframe', async () => {
+    // 7 assertions per lock created (1 additional for erc20)
+    expect.assertions(21) // Assertions inside dashboard.deployLock block!
```

Comment:
I am not sure that this is a relevant test to cover the user case of someone purchasing a key. While it does test the setup of this test I believe that we may be able to find a better place for this

Reply:
#4314 does what you suggest, take a look at the existing `adremover-loggedin.test.js`

https://github.com/unlock-protocol/unlock/pull/4314/files#diff-1a0b76ee27b9dfbedb6e485d1e16c313

It's just a shuffle,  as this line is required. Putting it in the `beforeAll` causes the test to fail